tests/drivers/i2s/i2s_api/src/test_i2s_errors.c

```diff
@@ -0,0 +1,168 @@
+
+	err = i2s_configure(dev_i2s, I2S_DIR_TX, &invalid_config);
+	zassert_not_equal(
+		err, 0,
+		"I2S configuration did not detect improper stream format (I2S_FMT_DATA_ORDER_LSB)");
+
+	invalid_config.format = I2S_FMT_DATA_FORMAT_I2S;
+	invalid_config.channels = 3U;
+	err = i2s_configure(dev_i2s, I2S_DIR_TX, &invalid_config);
+	zassert_not_equal(err, 0,
+			  "I2S configuration did not detect improper channels configuration (3)");
+
+	invalid_config.channels = 2U;
+	invalid_config.options = I2S_OPT_LOOPBACK | I2S_OPT_PINGPONG;
+	err = i2s_configure(dev_i2s, I2S_DIR_TX, &invalid_config);
+	zassert_not_equal(
+		err, 0,
+		"I2S configuration did not detect improper options (I2S_OPT_LOOPBACK | I2S_OPT_PINGPONG)");
+}
+
+ZTEST_USER(i2s_errors, test_i2s_config_attempt_in_wrong_state)
+{
+	int err;
+	int config_err;
+	char tx_data[BLOCK_SIZE];
+	struct i2s_config inactive_config = { .word_size = 16U,
+					      .channels = 2U,
+					      .format = I2S_FMT_DATA_FORMAT_I2S,
+					      .frame_clk_freq = FRAME_CLK_FREQ,
+					      .block_size = BLOCK_SIZE,
+					      .timeout = TIMEOUT,
+					      .options = I2S_OPT_FRAME_CLK_MASTER |
+							 I2S_OPT_BIT_CLK_MASTER,
+					      .mem_slab = &tx_mem_slab };
+
+	memset(tx_data, 0x00, sizeof(tx_data));
+
+	err = i2s_configure(dev_i2s, I2S_DIR_TX, &inactive_config);
+	zassert_equal(err, 0, "I2S interface configuration failed, err=%d", err);
+
+	err = i2s_buf_write(dev_i2s, tx_data, BLOCK_SIZE);
+	zassert_equal(err, 0, "I2S buffer write unexpected error: %d", err);
+
+	err = i2s_trigger(dev_i2s, I2S_DIR_TX, I2S_TRIGGER_START);
+	zassert_equal(err, 0, "I2S_TRIGGER_START unexpected error: %d", err);
+
+	config_err = i2s_configure(dev_i2s, I2S_DIR_TX, &inactive_config);
+
+	err = i2s_trigger(dev_i2s, I2S_DIR_TX, I2S_TRIGGER_STOP);
+	zassert_equal(err, 0, "I2S_TRIGGER_STOP unexpected error: %d", err);
+
+	zassert_not_equal(
+		config_err, 0,
+		"I2S configuration should not be possible in states other than I2S_STATE_READY %d",
+		err);
+}
+
+ZTEST_USER(i2s_errors, test_i2s_incorrect_trigger)
+{
+	int err;
+	char tx_data[BLOCK_SIZE];
+	struct i2s_config test_config = { .word_size = 16U,
+					  .channels = 2U,
+					  .format = I2S_FMT_DATA_FORMAT_I2S,
+					  .frame_clk_freq = FRAME_CLK_FREQ,
+					  .block_size = BLOCK_SIZE,
+					  .timeout = TIMEOUT,
+					  .options =
+						  I2S_OPT_FRAME_CLK_MASTER | I2S_OPT_BIT_CLK_MASTER,
+					  .mem_slab = &tx_mem_slab };
+
+	memset(tx_data, 0x00, sizeof(tx_data));
+
+	err = i2s_configure(dev_i2s, I2S_DIR_TX, &test_config);
+	zassert_equal(err, 0, "CFG err=%d", err);
+
+	err = i2s_buf_write(dev_i2s, tx_data, BLOCK_SIZE);
+	zassert_equal(err, 0, "I2S buffer write unexpected error: %d", err);
+
+	err = i2s_trigger(dev_i2s, I2S_DIR_TX, INVALID_TRIGGER_SETTING);
+	zassert_equal(err, -EINVAL, "I2S invalid trigger setting not detected: err=%d", err);
+}
+
+ZTEST_USER(i2s_errors, test_i2s_unconfigured_access)
+{
+	int err;
+	char tx_data[BLOCK_SIZE];
```

Comment:
This will raise static analysis error because the buffer is used uninitialized here.